src/scripts/main.js

```diff
@@ -1,5 +1,28 @@
 'use strict';
 
 document.addEventListener('click', (e) => {
-  // write code here
+  if (e.target.className === 'wall') {
+    const x = e.clientX;
+    const y = e.clientY;
+    const field = e.target.getBoundingClientRect();
+    const spider = document.querySelector('.spider');
+
+    const borderX = (e.target.offsetWidth - e.target.clientWidth) / 2;
+    const borderY = (e.target.offsetHeight - e.target.clientHeight) / 2;
+
+    const maxLeft = field.width - spider.offsetWidth - borderX * 2;
+    const maxTop = field.height - spider.offsetHeight - borderY * 2;
+
+    let spiderLeft = x - field.left - spider.offsetWidth / 2 - borderX;
+    let spiderTop = y - field.top - spider.offsetHeight / 2 - borderY;
+
+    spiderLeft = spiderLeft < 0 ? 0 : spiderLeft;
+    spiderLeft = spiderLeft > maxLeft ? maxLeft : spiderLeft;
```

Comment:
These lines are a bit repetitive and could be simplified by using the `Math.min` and `Math.max` functions. You could set `spiderLeft` to `Math.max(0, Math.min(spiderLeft, maxLeft))`, which would ensure that `spiderLeft` is never less than 0 and never more than `maxLeft`. The same applies to `spiderTop`.